### ISSUE TITLE
Tray icon not showing

### ISSUE THREAD

**Issue by ghost** [mono/mono#12202]
Mono applications that are minimized to try, does not show the real tray icon but a fully-black tray icon instead.
This bug was first reported [in the old bugzilla tracker](https://bugzilla.xamarin.com/show_bug.cgi?id=8935), In particular affects [KeePass](https://keepass.info/), and the KeePass developers asked multiple times to report this to the Mono developers ([1](https://sourceforge.net/p/keepass/bugs/1051/), [2](https://sourceforge.net/p/keepass/discussion/329221/thread/09c1a6a5/),[3](https://sourceforge.net/p/keepass/discussion/329221/thread/1f1c2c7b/))

## Steps to Reproduce

1. Download KeePass from  [here](https://keepass.info/),
2.  Run `mono KeePass.exe`
3. A new systray icon is created, but it is full black instead of the KeePass icon

## On which platforms did you notice this

[ ] macOS
[X ] Linux
[ ] Windows

**Version Used**:

```
 mono --version
Mono JIT compiler version 4.6.2 (Debian 4.6.2.7+dfsg-1ubuntu1)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug 
        LLVM:          supported, not enabled.
        GC:            sgen
```

**Comment by DerSandmann-Badcode** [comment]
Figured; As far as I know, Ubuntu doesn't support TrayIcons from Winforms anymore, you must use Appindicator. So many Windows apps that previously worked with a TrayIcon will just show a black square.

**Comment by raffam** [comment]
So what are the steps I have to follow in order to fix the problem?

**Comment by DerSandmann-Badcode** [comment]
I don't think we can really fix this... TrayIcons are not handled correctly in Ubuntu anymore. It's an issue with Mono\Ubuntu.

1. Fix Wine\Ubuntu's Implementation of TrayIcon - Currently TrayIcons will show either a black square, or 'Copy' the icon of another application

![image](https://user-images.githubusercontent.com/8218786/50542689-65f08f00-0c28-11e9-84aa-94c6aa82ac47.png)


2. Rewrite the Application (KeePass) in GTK or something similar that supports AppIndicator

There's C# bindings for AppIndicator in GTK
![image](https://user-images.githubusercontent.com/8218786/50542682-317cd300-0c28-11e9-99dc-d3e09295a0e6.png)
